adapters/smaato/banner_test.go

```diff
@@ -0,0 +1,42 @@
+package smaato
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestExtractAdmBanner(t *testing.T) {
+	tests := []struct {
+		testName         string
+		adMarkup         string
+		curls            []string
+		expectedAdMarkup string
+	}{
+		{
+			testName:         "extract banner without curls",
+			adMarkup:         `<a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a>`,
+			expectedAdMarkup: `<div style="cursor:pointer" ><a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a></div>`,
+			curls:            []string{},
+		},
+		{
+			testName:         "extract banner with nil curls",
+			adMarkup:         `<a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a>`,
+			expectedAdMarkup: `<div style="cursor:pointer" ><a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a></div>`,
+			curls:            nil,
+		},
+		{
+			testName:         "extract banner with curls",
+			adMarkup:         `<a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a>`,
+			expectedAdMarkup: `<div style="cursor:pointer" onclick=fetch(decodeURIComponent('curls.net'.replace(/\+/g, ' ')), {cache: 'no-cache'});><a rel="nofollow" href="https://prebid.net/click"><img src="https://prebid.net/images/image.png" alt="" width="480" height="320" /></a></div>`,
+			curls:            []string{"curls.net"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			adMarkup := extractAdmBanner(tt.adMarkup, tt.curls)
+
+			assert.Equal(t, tt.expectedAdMarkup, adMarkup)
+		})
+	}
+}
```

Comment:
This is fine, but doesn't count towards the test coverage for our review purposes. Please ensure these cases are covered with the json test framework.

Reply:
good catch, I added one more test case `curls=nil`
highly unlikely that we would ever return this from the server, but better have it covered